lib/Model/Binary.cpp

```diff
@@ -366,6 +366,28 @@ bool Segment::verify(VerifyHelper &VH) const {
       return VH.fail("Invalid relocation", Relocation);
   }
 
+  if (not Type().empty()) {
+
+    if (not Type().isValid())
+      return VH.fail("Invalid segment type", *this);
+
+    // The segment has a type
+
+    auto *Struct = dyn_cast<model::StructType>(Type().get());
+    if (not Struct)
+      return VH.fail("The segment type is not a StructType", *this);
+
+    if (VirtualSize() != Struct->Size()) {
+      return VH.fail(Twine("VirtualSize: ") + Twine(VirtualSize())
+                       + Twine(" != StackFrameType->Size: ")
+                       + Twine(Struct->Size()),
+                     *this);
```

Comment:
These are errors we expose to end users, let's use sentences, not C-like expressions.